### ISSUE TITLE
performance unstationality

### ISSUE THREAD

**Issue by tarai2** [tarai2/Numer-ai#8]
https://github.com/tarai2/Numer-ai/wiki/EraBoosting  
https://github.com/tarai2/Numer-ai/wiki/Performance-Stationarity-and-Smart-Sharpe  
上記wikiにもまとめたようにパフォーマンスの不均一性が問題となっている。
パフォーマンスの均一化には方針が2つあると思っていて、  
1. 局面判別器+局面限定モデルといった、複数のモデルを作成する
2. なんらかの工夫を施した、1つの優れたモデルで対応する

**Comment by tarai2** [comment]
### Feature Importaneを利用したEraのクラスタリング

**Comment by tarai2** [comment]
### 優れた1つのモデルを作るためのアイデア
- rank corr  
yとy_hatの2つがバッチで必要. 最大化したい. ROS氏は採用しているようだ.

- smart sharpe  
y_hatが時系列で必要. 最大化したい.

- 適用対象:
    - NN
    - Tree系

- 適用方法:
    - monitoring指標として
    - ロス関数として

**Comment by tarai2** [comment]
・Featureに対してModel直交化=>パフォーマンス安定
(https://github.com/tarai2/Numer-ai/wiki/Model-Diagnostics:-Feature-Exposure)
・NNのtrainは特にlossの設計が大事
(https://github.com/tarai2/Numer-ai/wiki/NNをTournamentデータにフィットすることの難しさについて)  
・**2つのmodelをuncorrelatedに学習させる**
(https://forum.numer.ai/t/learning-two-uncorrelated-models/400/16)  
・torchを使ってxgbでカスタムlossを用いる方法
(https://forum.numer.ai/t/custom-loss-functions-for-xgboost-using-pytorch/960)  

以上のポストから察するに、  
**uncorrelatedなモデル, FeatureExposureの低いモデルを複数作る => アンサンブルして強いモデルに**  
が正解なのではないか。この方針で行きたい。
